### ISSUE TITLE
Invalid database name with MongoDB and replica set

### ISSUE THREAD

**Issue by andreoav** [typeorm/typeorm#6482]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[x] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[x] `0.2.25` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

After upgrading to version `0.2.25` my application can't connect to MongoDB.

`Invalid database name: 'siteDatabase?readPreference=primary&replicaSet=RS&authSource=siteDatabase'`

On production environments I have a replica set and the URI is like the following.

```
mongodb://user:password@db-mongo-PRD-siteDatabase01.example.com:27017,db-mongo-PRD-siteDatabase02.example.com:27017/siteDatabase?readPreference=primary&replicaSet=RS&authSource=siteDatabase
```

**Comment by scadgek** [comment]
+1, I ran into the same problem after upgrading to 0.2.25.

**Comment by rcursaru** [comment]
+1, solved by reverting to 0.2.24
scary thing, 0.2.25 is out since 19th May...

**Comment by imnotjames** [comment]
More information needed - can you include how you are specifying this for TypeORM so I can replicate locally?

**Comment by imnotjames** [comment]
This should be fixed as of 0.2.26 - This is a duplicate of issue #6389 which was fixed by #6390 !

**Comment by scadgek** [comment]
Awesome, thank you James!
